From 452df9a9814ea2d42a01ec838c537059e1e182d0 Mon Sep 17 00:00:00 2001
From: Zhihai Xu <zxu@apache.org>
Date: Sat, 19 Sep 2015 00:10:06 -0700
Subject: [PATCH 1260/2748] MAPREDUCE-6460.
 TestRMContainerAllocator.testAttemptNotFoundCausesRMCommunicatorException
 fails. Contributed by Zhihai Xu.

(cherry picked from commit e3ace31e2b7e3a8dc6cf3dc044d5ef2a9aa12e29)

Change-Id: I0cf8b86dc9be5e9bf1e7a9815b8aafe476655126
---
 .../v2/app/rm/TestRMContainerAllocator.java        |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/rm/TestRMContainerAllocator.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/rm/TestRMContainerAllocator.java
index f7498b9..8331041 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/rm/TestRMContainerAllocator.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/rm/TestRMContainerAllocator.java
@@ -112,6 +112,7 @@
 import org.apache.hadoop.yarn.server.resourcemanager.recovery.MemoryRMStateStore;
 import org.apache.hadoop.yarn.server.resourcemanager.recovery.RMStateStore;
 import org.apache.hadoop.yarn.server.resourcemanager.rmapp.RMApp;
+import org.apache.hadoop.yarn.server.resourcemanager.rmapp.RMAppState;
 import org.apache.hadoop.yarn.server.resourcemanager.scheduler.Allocation;
 import org.apache.hadoop.yarn.server.resourcemanager.scheduler.ResourceScheduler;
 import org.apache.hadoop.yarn.server.resourcemanager.scheduler.event.SchedulerEvent;
@@ -2546,7 +2547,7 @@ public void testAttemptNotFoundCausesRMCommunicatorException()
 
     // Now kill the application
     rm.killApp(app.getApplicationId());
-
+    rm.waitForState(app.getApplicationId(), RMAppState.KILLED);
     allocator.schedule();
   }
 
-- 
1.7.9.5

